### PR TITLE
Tag searches

### PR DESCRIPTION
**Pull request by huwshimi**
Clicking tags takes you to search results for that tag.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- view `/:flags:/react/`
- open the mid-point
- click on a tag at the bottom of the mid-point
- you should get a URL something like `?search&tags=applications` and a list of charms

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3181/
Test PASSed.

**Comment by anthonydillon** [comment]
Thanks @huwshimi QA and code :+1:

File: jujugui/static/gui/src/app/views/state.js [hunk 1]
```diff
@@ -368,9 +371,22 @@ YUI.add('juju-app-state', function(Y) {
       if (window.flags && window.flags.react) {
         if (Object.keys(queryValues).length > 0) {
           url = url.replace(/\/$/, '');
-          Object.keys(queryValues).forEach((key) => {
-            url += '?' + key + '=' + queryValues[key];
-          });
+          var keys = Object.keys(queryValues);
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/state.js]: Thanks for fixing this :)

File: jujugui/static/gui/src/app/views/state.js [hunk 2]
```diff
@@ -368,9 +371,22 @@ YUI.add('juju-app-state', function(Y) {
...
+          if (keys.length > 0) {
+            url += '?';
+            var i = 0;
+            keys.forEach((key) => {
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/state.js]: `forEach` passes the index as the second param which could be used instead of `i`

**Comment by hatched** [comment]
Lookin good!

:+1: With a trivial

File: jujugui/static/gui/src/app/views/state.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/state.js]: Ah, I did not know that. Thanks!

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3189/
Test FAILed.

**Comment by huwshimi** [comment]
Thanks @hatched and @anthonydillon :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3190/
Test PASSed.

**Pull request merged**